core/tooltip.js

```diff
@@ -234,235 +32,421 @@ const getTargetObject = function(obj) {
+     * appears).
+     * @protected
+     */
+    this.lastX = 0;
+
+    /**
+     * Last observed Y location of the mouse pointer (freezes when tooltip
+     * appears).
+     * @protected
+     */
+    this.lastY = 0;
+
+    /**
+     * Current element being pointed at.
+     * @type {?Element}
+     * @protected
+     */
+    this.element = null;
+
+    /**
+     * Once a tooltip has opened for an element, that element is 'poisoned' and
+     * cannot respawn a tooltip until the pointer moves over a different
+     * element.
+     * @type {?Element}
+     * @protected
+     */
+    this.poisonedElement = null;
+
+    /**
+     * Maximum width (in characters) of a tooltip.
+     * @alias Blockly.Tooltip.LIMIT
+     */
+    this.LIMIT = 50;
+
+    /**
+     * Horizontal offset between mouse cursor and tooltip.
+     * @alias Blockly.Tooltip.OFFSET_X
+     */
+    this.OFFSET_X = 0;
+
+    /**
+     * Vertical offset between mouse cursor and tooltip.
+     * @alias Blockly.Tooltip.OFFSET_Y
+     */
+    this.OFFSET_Y = 10;
+
+    /**
+     * Radius mouse can move before killing tooltip.
+     * @alias Blockly.Tooltip.RADIUS_OK
+     */
+    this.RADIUS_OK = 10;
+
+    /**
+     * Delay before tooltip appears.
+     * @alias Blockly.Tooltip.HOVER_MS
+     */
+    this.HOVER_MS = 750;
+
+    /**
+     * Horizontal padding between tooltip and screen edge.
+     * @alias Blockly.Tooltip.MARGINS
+     */
+    this.MARGINS = 5;
+
+    /**
+     * The HTML container.  Set once by createDom.
+     * @type {?Element}
+     * @protected
+     */
+    this.DIV = null;
   }
-  // Create an HTML container for popup overlays (e.g. editor widgets).
-  DIV = document.createElement('div');
-  DIV.className = 'blocklyTooltipDiv';
-  const container = common.getParentContainer() || document.body;
-  container.appendChild(DIV);
-};
-exports.createDom = createDom;
 
-/**
- * Binds the required mouse events onto an SVG element.
- * @param {!Element} element SVG element onto which tooltip is to be bound.
- * @alias Blockly.Tooltip.bindMouseEvents
- */
-const bindMouseEvents = function(element) {
-  element.mouseOverWrapper_ =
-      browserEvents.bind(element, 'mouseover', null, onMouseOver);
-  element.mouseOutWrapper_ =
-      browserEvents.bind(element, 'mouseout', null, onMouseOut);
-
-  // Don't use bindEvent_ for mousemove since that would create a
-  // corresponding touch handler, even though this only makes sense in the
-  // context of a mouseover/mouseout.
-  element.addEventListener('mousemove', onMouseMove, false);
-};
-exports.bindMouseEvents = bindMouseEvents;
-
-/**
- * Unbinds tooltip mouse events from the SVG element.
- * @param {!Element} element SVG element onto which tooltip is bound.
- * @alias Blockly.Tooltip.unbindMouseEvents
- */
-const unbindMouseEvents = function(element) {
-  if (!element) {
-    return;
+  /**
+   * Returns the tooltip text for the given element.
+   * @param {?Object} object The object to get the tooltip text of.
```

Comment:
No, because it's allowed that the object doesn't have a tooltip property. (in that case getTargetObject would return null and this function would return an empty string so no tooltip is shown)